File: 0231-power-of-two/python/Solution.py
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        while n != 0:
            if (n & 1) == 1:
                if (n >> 1) == 0:
                    return True
                else:
                    return False
            n >>= 1
        return False

File: 0231-power-of-two/cpp/Solution.cpp
class Solution {
public:
    bool isPowerOfTwo(int n) {
        while (n != 0) {
            if ((n & 1) == 1) {
                if ((n >> 1) == 0) {
                    return true;
                } else {
                    return false;
                }
            }
            n >>= 1;
        }
        return false;
    }
};
